adagrams/game.py

```diff
@@ -1,13 +1,162 @@
+        #Access an element within letter_pool list, determined by random_index.
+        random_letter = letter_pool[random_index]
+        #Add the newly generated random letter to letter_hand list.
+        letter_hand.append(random_letter)
+        #Remove the random_letter from letter_pool
+        letter_pool.remove(random_letter)
+
+    #Return letter_hand list of 10 letters.
+    return letter_hand
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    word = word.upper()
+    #Create a dictionary for word to store the word and its frequency.
+    word_count = {}
+    #Iterate through each element in word and add it to word_count dictionary.
+    for char in word:
+        if char in word_count:
+            word_count[char] += 1
+        else:
+            word_count[char] = 1
+    
+
+    #Create a dictionary for letter_bank to store each word and its frequency.
+    letter_bank_count = {}
+    #Iterate through each element in letter_bank list and add it to letter_bank_count dictionary.
+    for i in letter_bank:
+        if i in letter_bank_count:
+            letter_bank_count[i] += 1
+        else:
+            letter_bank_count[i] = 1
+
+    
+    #For every letter in word_count, the letter in letter_bank_count is greater than or equal to it.
+    #Iterate through word_count dictionary
+    for char in word_count.keys():
+        if char not in letter_bank_count or word_count[char] > letter_bank_count[char]:
+            return False
+    return True
+        
+
 
 def score_word(word):
-    pass
+    word = word.upper()
+    #Create a dictionary to store score chart.
+    letter_values = {
+        'A': 1,
+        'B': 3,
+        'C': 3,
+        'D': 2,
+        'E': 1,
+        'F': 4,
+        'G': 2,
+        'H': 4,
+        'I': 1,
+        'J': 8,
+        'K': 5,
+        'L': 1,
+        'M': 3,
+        'N': 1,
+        'O': 1,
+        'P': 3,
+        'Q': 10,
+        'R': 1,
+        'S': 1,
+        'T': 1,
+        'U': 1,
+        'V': 4,
+        'W': 4,
+        'X': 8,
+        'Y': 4,
+        'Z': 10
+    }
+    word_score = 0
+
+
+    #Iterate through word string
+    for char in word:
+    #Match each char in word to element in letter_values
+        letter_value = letter_values[char]
+        #Sum the value of element(value) of letter_values and store in word_score
+        word_score += letter_value
+        #If the length of word is 7,8,9,10 - add 8 to word_score
+    if len(word) == 7 or len(word) == 8 or len(word) == 9 or len(word) == 10:
+        word_score += 8
+    return word_score
+
+
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    best_score = 0
+    best_scoring_tuple = None
+    #Iterate through every word in word_list.
+    for word in word_list:
+        #Get the score of each word.
+        word_score = score_word(word)
+        #Add word and it's score to tuple
+        word_score_tuple = (word, word_score)
```

Comment:
Like the case with `best_scoring_tuple`, you are storing data twice in memory (`word` in this case).